static/js/notifications.js

```diff
@@ -255,7 +255,11 @@ function process_notification(notification) {
     // Convert the content to plain text, replacing emoji with their alt text
     content = $('<div/>').html(message.content);
     ui.replace_emoji_with_text(content);
-    content = content.text();
+
+    // Prevent empty notification when message contains only an image
+    content = (content.text() === '' && content.children().first().hasClass("message_inline_image"))
+            ? title + " sent you an image"
```

Comment:
We should tag this for translation; see zulip.readthedocs.io/en/latest/translating/index.html.  We might need to do something a bit fancy here; @umairwaheed can help if need be.

Also, I'd prefer an if statement block for this than the fancy `? :` syntax: it's more readable and also our test coverage tools will notice if one of the code paths isn't tested.